File: TungstenCore/Models/Segment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TungstenCore.Models
{
    public class Segment : Entity<string>
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();
        [Display(Name = "Namn")]
        public string Name { get; set; }
        [Display(Name = "Beskrivning")]
        public string Description { get; set; }
        
        [Required, ForeignKey("Course")]
        public string CourseId { get; set; }
        public Course Course { get; set; }

        public virtual ICollection<Assignment> Assignments { get; set; }
    }
}

File: TungstenCore/ClientApp/app/classes/Segment.ts
import { ISegment } from '../interfaces/segment';
import { ICourse } from '../interfaces/course';
export class Segment implements ISegment{
    Id: string;
    Name: string;
    Description: string;
    CourseId: string;
    Course: ICourse;

    constructor(name: string, description: string, courseid: string) {
        this.Name = name;
        this.Description = description;
        this.CourseId = courseid;
    }
}